weaveworks/weave-gitops-enterprise
# Don't show clusters and templates permission errors if viewing apps/sources

- Refactors Clusters and Templates contexts into hooks
  - As the selectedClusters state (which clusters to delete) is no longer in a common context between the deleteClusterDialog and the clustersList we do a small refactor and clear the selectedClusters onDialogClose instead of having the Dialog do that itself.

Fixes #1422 

### To test:

Pre-reqs:
- Be part of the `wge` group: https://gitlab.git.dev.weave.works/wge/

Test
- Login to https://35.188.40.143:30080/ with OIDC + gitlab, navigate to apps/sources, wait ~20s, see the error toast about no gitopscluster/capitemplates permissions. 
- Checkout this branch locally and run: `PROXY_HOST=https://35.188.40.143:30080/ yarn start`
- Login with OIDC + gitlab, **The redirect will be incorrect and needs correction**, when you see an error in the browser, edit the URL manually, replacing `https://35.188.40.143:30080/` with `http://localhost:3000` and hit <kbd>enter</kbd> and you should be logged in locally.
- Notice no more errors on the apps/sources page.
